### PR TITLE
[AU4] Double-click on track/clip behaviour

### PR DESCRIPTION
**Pull request by grliszas14**
Resolves: https://github.com/audacity/audacity/issues/7474
Resolves: https://github.com/audacity/audacity/issues/7473

### CODE REVIEW

File: au4/src/au3wrap/internal/domaccessor.h [hunk 1]
```diff
@@ -19,6 +19,7 @@ class DomAccessor
     static WaveTrack* findWaveTrack(AudacityProject& prj, const TrackId& au3trackId);
     static std::shared_ptr<WaveClip> findWaveClip(WaveTrack* track, uint64_t au3ClipId);
     static std::shared_ptr<WaveClip> findWaveClip(AudacityProject& prj, const TrackId& au3trackId, size_t index);
+    static std::shared_ptr<WaveClip> findWaveClipAtGivenTime(AudacityProject &prj, const trackedit::TrackId &trackId, trackedit::secs_t time);
```
Comment (Eism) [inline au4/src/au3wrap/internal/domaccessor.h:22]: I think we can name it just `findWaveClip`
"AtGivenTime" can be understood from the input params

File: au4/src/projectscene/view/clipsview/selectionviewcontroller.cpp [hunk 2]
```diff
@@ -107,6 +107,19 @@ void SelectionViewController::onSelectionDraged(double x1, double x2, bool compl
     selectionController()->setDataSelectedEndTime(m_context->positionToTime(x2, true /*withSnap*/), completed);
 }
 
+void SelectionViewController::selectTrackAudioData(double y)
+{
+    const std::vector<TrackId> tracks = determinateTracks(m_startPoint.y(), y);
+    selectionController()->setSelectedTrackAudioData(tracks.at(0));
```
Comment (Eism) [inline au4/src/projectscene/view/clipsview/selectionviewcontroller.cpp:113]: can we replace this with `selectionController()->setSelectedTrack(trackId)`?

File: au4/src/projectscene/view/clipsview/selectionviewcontroller.cpp [hunk 3]
```diff
@@ -107,6 +107,19 @@ void SelectionViewController::onSelectionDraged(double x1, double x2, bool compl
...
+}
+
+void SelectionViewController::selectClipAudioData(double x, double y)
+{
+    const std::vector<TrackId> tracks = determinateTracks(m_startPoint.y(), y);
+    const secs_t time = m_context->positionToTime(x);
+    selectionController()->setSelectedClipAudioData(tracks.at(0), time);
```
Comment (Eism) [inline au4/src/projectscene/view/clipsview/selectionviewcontroller.cpp:120]: and this with `selectionController()->setSelectedClip(clipId)`?

File: au4/src/projectscene/view/clipsview/selectionviewcontroller.cpp (hunk 2 above)
Comment (grliszas14) [reply au4/src/projectscene/view/clipsview/selectionviewcontroller.cpp:113]: not quite: right here we want to precisely select all of the audio data that belongs to the track. The function you mention just marks that currently track with particular trackId was clicked for example (or that you selected part of that data, etc.)

File: au4/src/projectscene/view/clipsview/selectionviewcontroller.cpp (hunk 3 above)
Comment (grliszas14) [reply au4/src/projectscene/view/clipsview/selectionviewcontroller.cpp:120]: same here: we don't want to select a clip, we want to select all audio data that belongs to the clip

File: au4/src/au3wrap/internal/domaccessor.h (hunk 1 above)
Comment (grliszas14) [reply au4/src/au3wrap/internal/domaccessor.h]: fixed

**Review by Eism — APPROVED** [review]

**Review by Eism — APPROVED** [review]

**Pull request merged**